runtime/Cpp/runtime/src/misc/IntervalSet.h

```diff
@@ -93,82 +106,80 @@ namespace misc {
      *
      * @return the maximum value contained in the set. If the set is empty, this
      * method returns {@link Token#INVALID_TYPE}.
      */
-    virtual ssize_t getMaxElement() const;
+    ssize_t getMaxElement() const;
 
     /**
      * Returns the minimum value contained in the set.
      *
      * @return the minimum value contained in the set. If the set is empty, this
      * method returns {@link Token#INVALID_TYPE}.
      */
-    virtual ssize_t getMinElement() const;
+    ssize_t getMinElement() const;
 
     /// <summary>
     /// Return a list of Interval objects. </summary>
-    virtual std::vector<Interval> getIntervals() const;
+    std::vector<Interval> const& getIntervals() const;
 
-    virtual size_t hashCode() const;
+    size_t hashCode() const;
 
     /// Are two IntervalSets equal?  Because all intervals are sorted
     ///  and disjoint, equals is a simple linear walk over both lists
     ///  to make sure they are the same.
     bool operator == (const IntervalSet &other) const;
-    virtual std::string toString() const;
-    virtual std::string toString(bool elemAreChar) const;
+    std::string toString() const;
+    std::string toString(bool elemAreChar) const;
 
     /**
      * @deprecated Use {@link #toString(Vocabulary)} instead.
      */
-    virtual std::string toString(const std::vector<std::string> &tokenNames) const;
-    virtual std::string toString(const dfa::Vocabulary &vocabulary) const;
+    std::string toString(const std::vector<std::string> &tokenNames) const;
+    std::string toString(const dfa::Vocabulary &vocabulary) const;
 
   protected:
     /**
      * @deprecated Use {@link #elementName(Vocabulary, int)} instead.
      */
-    virtual std::string elementName(const std::vector<std::string> &tokenNames, ssize_t a) const;
-    virtual std::string elementName(const dfa::Vocabulary &vocabulary, ssize_t a) const;
+    std::string elementName(const std::vector<std::string> &tokenNames, ssize_t a) const;
+    std::string elementName(const dfa::Vocabulary &vocabulary, ssize_t a) const;
 
   public:
-    virtual size_t size() const;
-    virtual std::vector<ssize_t> toList() const;
-    virtual std::set<ssize_t> toSet() const;
+    size_t size() const;
+    std::vector<ssize_t> toList() const;
+    std::set<ssize_t> toSet() const;
 
     /// Get the ith element of ordered set.  Used only by RandomPhrase so
     /// don't bother to implement if you're not doing that for a new
     /// ANTLR code gen target.
-    virtual ssize_t get(size_t i) const;
-    virtual void remove(size_t el); // For mapping of e.g. Token::EOF to -1 etc.
-    virtual void remove(ssize_t el);
-    virtual bool isReadOnly() const;
-    virtual void setReadOnly(bool readonly);
+    ssize_t get(size_t i) const;
+    void remove(size_t el); // For mapping of e.g. Token::EOF to -1 etc.
+    void remove(ssize_t el);
 
   private:
-    void InitializeInstanceFields();
+    void addItems() { /* No-op */ }
```

Comment:
This one seems strange. Can this be removed?

Reply:
No, this is necessary for the variadic template constructor.

Reply:
Ah right, to end the iteration. OK.